src/AppInstallerCLICore/ChannelStreams.h

```diff
@@ -11,36 +11,6 @@
 
 namespace AppInstaller::CLI::Execution
 {
-    namespace details
-    {
-        // List of approved types for output, others are potentially not localized.
-        template <typename T>
-        struct IsApprovedForOutput
-        {
-            static constexpr bool value = false;
-        };
-
-#define WINGET_CREATE_ISAPPROVEDFOROUTPUT_SPECIALIZATION(_t_) \
-        template <> \
-        struct IsApprovedForOutput<_t_> \
-        { \
-            static constexpr bool value = true; \
-        }
-
-        // It is assumed that single char values need not be localized, as they are matched
-        // ordinally or they are punctuation / other.
-        WINGET_CREATE_ISAPPROVEDFOROUTPUT_SPECIALIZATION(char);
-        // Localized strings (and from an Id for one for convenience).
-        WINGET_CREATE_ISAPPROVEDFOROUTPUT_SPECIALIZATION(Resource::StringId);
-        WINGET_CREATE_ISAPPROVEDFOROUTPUT_SPECIALIZATION(Resource::LocString);
-        // Strings explicitly declared as localization independent.
-        WINGET_CREATE_ISAPPROVEDFOROUTPUT_SPECIALIZATION(Utility::LocIndView);
-        WINGET_CREATE_ISAPPROVEDFOROUTPUT_SPECIALIZATION(Utility::LocIndString);
-        // Normalized strings come from user data and should therefore already by localized
-        // by how they are chosen (or there is no localized version).
-        WINGET_CREATE_ISAPPROVEDFOROUTPUT_SPECIALIZATION(Utility::NormalizedString);
-    }
-
     // The base stream for all channels.
     struct BaseStream
     {

```

Comment:
Good point, I will update the TODO message here. Unfortunately, uncommenting this `static_assert` will require a completely new set of changes, but it's certainly doable (maybe in several smaller PRs?). That said, with the new localization syntax introduced in this PR, this rule will be enforced on all localized strings placeholders, which is a good step in solving part of the problem, and ultimately it will limit the number of changes required to uncomment this in the future.